### PR TITLE
[MAIN-352] Add Clock Syncing Server Endpoint

### PR DESCRIPTION
**Pull request by sarahannali**

### CODE REVIEW

File: services/server/test/integration/serviceInstance.js [hunk 1]
```diff
@@ -31,6 +31,22 @@ test('GET /instance returns current serviceInstance', async (t) => {
...
+  const requestTimestampMS = Date.now();
+  const { data: { date }, status } = await api.get('/instance/date');
+  t.is(status, StatusCodes.OK);
+
+  const bufferMS = 500;
+
+  const lowerBound = requestTimestampMS;
+  const upperBound = requestTimestampMS + bufferMS;
+
+  t.truthy(date >= lowerBound);
```
Comment (kevo1ution) [inline services/server/test/integration/serviceInstance.js:46]: `t.true` seems like a better fit here. Also would be nice for the assertion string to include the values of `date, lowerBound, and upperBound` (e.g. `t.true(bool, "date=, lowerBound=, upperBound="`)

File: services/server/src/models/serviceInstance/route.js [hunk 2]
```diff
@@ -19,6 +19,10 @@ function setupRouter({ io, serviceInstanceId }) {
       res.status(StatusCodes.OK).json({ serviceInstance: serviceInstance.toJSON() });
     }));
 
+  router.get('/date', asyncHandler(async (req, res) => {
+    res.status(StatusCodes.OK).json({ date: Date.now() });
```
Comment (kevo1ution) [inline services/server/src/models/serviceInstance/route.js:23]: Can we return this as an ISO timestamp? Even though it adds extra parsing (ISO is supported by both cleint and server), it makes it easier to debug whats going on.

**Review by kevo1ution — CHANGES_REQUESTED** [review]

File: services/server/src/models/serviceInstance/route.js (hunk 2 above)
Comment (sarahannali) [reply services/server/src/models/serviceInstance/route.js]: done

File: services/server/test/integration/serviceInstance.js (hunk 1 above)
Comment (sarahannali) [reply services/server/test/integration/serviceInstance.js]: done

**Review by kevo1ution — APPROVED** [review]
LGTM! Let get this mfkinbread. I'll merge after builds pass

**Pull request merged**